src/js/game/GameController.js

```diff
@@ -1238,6 +1238,10 @@ class GameController {
           commandQueueItem.succeeded();
         });
       });
+    } else if (frontBlock.isRail) {
+      player.onTracks = true;
```

Comment:
any reason why this isn't inside `playTrack`?

Reply:
Because playTrack is called multiple times, and we only need to set it once when we hop on and once when we end.

It certainly COULD be in playTrack but we'd either do a bunch of setting to true when already true, or have an extra if statement to check if it's not already true?